src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -308,18 +308,23 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                     if (!refreshAll)
                     {
-                        applicationSettings = new Dictionary<string, ConfigurationSetting>(_applicationSettings, StringComparer.OrdinalIgnoreCase);
+                        mappedData = new Dictionary<string, ConfigurationSetting>(_mappedData, StringComparer.OrdinalIgnoreCase);
                         watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>(_watchedSettings);
 
                         foreach (KeyValueChange change in keyValueChanges.Values.Concat(changedKeyValuesCollection))
                         {
                             if (change.ChangeType == KeyValueChangeType.Deleted)
                             {
-                                applicationSettings.Remove(change.Key);
+                                mappedData.Remove(change.Key);
                             }
                             else if (change.ChangeType == KeyValueChangeType.Modified)
                             {
-                                applicationSettings[change.Key] = change.Current;
+                                ConfigurationSetting setting = change.Current;
+                                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.Mappers)
+                                {
+                                    setting = await func(setting).ConfigureAwait(false);
+                                }
+                                mappedData[change.Key] = setting;

```

Comment:
Check for null `setting` before updating cache. If its null, delete the key from mappedData.